### DISCUSSION TITLE
Adding more rule severity

### DISCUSSION THREAD

**Discussion by emadecode** [eslint/eslint#16769] [Ideas] [answered]
**What is the problem?**
When we add or create new rules, It's very common that other developers face some issues. Especially when git prevent to commit change due to ESLint error.

**Note:** I'm not talking about regular error that needs to be fixed. I'm talking about potential mistakes and bugs when we create custom rules.

**What is the idea?**
Adding more severity to the current [severities](https://eslint.org/docs/latest/user-guide/configuring/rules#rule-severities ).
Let's say we add a `Experimental` severity with id `3`. The Idea is to set certain rules between developers to report problems and fix issues before we change `Experimental` to `Error`. We won't stop developers from committing when this is in `Experimental` phase. We just show a message. And it can be reported if developer think the rule is acting weird. But after a while (Agreed time) we will change that to `Error` and developers should have resolved the issues meantime.
 In this way, Development process won't be interrupted by new rules.

What you think guys? Is this a good approach or do we have any similar approach currently?

**Comment by bmish** [comment]
That use case sounds like the existing `warn` severity.

**Reply by emadecode** [reply]
It's not the same. The `warn` severity has different meaning between developers and most use case is "**Better to not do that**". And usually there is no limit for committing the code with `warn` messages. But what i mean of `ExperimentalError` is "**you should avoid of doing that**" and resolve the problem ASAP. (either change the code or fix the linter bug)
Because after a while (a limited time) it will change to `Error` and you won't be able to commit your changes.

**Reply by sam3k** [reply]
Hi @emadecode, I second @bmish; `warn` is what this sounds like to me. "Better to not do that" is a warning; just like "you should avoid doing that". These are warnings that X action could lead to "some" consequence; but not necessarily an error. Also, deprecation messages are usually warnings.

**Reply by moniuch** [reply]
I second @emadecode. I would have made a good use of it during my administrative chores, but only if I had a way in the config to specify whether this severity of `3` should have an outcome of `1` or `2` (ie. should it exit process with zero or non-zero code, should it counts towards `--max-warnings` limit). That would be a great tool in my toolbox.

**Comment by nzakas** [comment, marked as answer]
Thanks for the suggestion @emadecode. At this point, there are no plans to add additional severities. While we may revisit this in the future, for the time being you'll need to use `warn`.
